sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -79,7 +80,33 @@ class GpuHashAggregateMeta(
         case "partial" => if (hashAggMode.contains(Final) || hashAggMode.contains(Complete)) {
           // replacing only Partial hash aggregates, so a Final or Complete one should not replace
           willNotWorkOnGpu("Replacing Final or Complete hash aggregates disabled")
-         }
+        }
+          // In partial mode, if there are non distinct functions and multiple distinct functions
+          // non distinct functions are computed using First operator. The final result would be
+          // incorrect for non distinct functions for partition size > 1. Reason for this is - if
+          // the first batch computed and sent to CPU doesn't contain all the rows required to
+          // comput non distinct function(s), then Spark would consider that value as final result
+          // (due to First)Falling back to CPU for this special case.
+          if (agg.aggregateExpressions.exists(e => e.aggregateFunction.isInstanceOf[First])) {
+            agg.aggregateExpressions.foreach(e => {
+              // Check if there is an `If` within `First`. This is included in the plan only when
+              // multiple distinct is present in the query.  We fall back to CPU in this case when
+              // the references of `If` is an aggregate.
+              if (e.aggregateFunction.isInstanceOf[First]) {
+                val t = e.aggregateFunction.asInstanceOf[First].child.isInstanceOf[If]
+                if (t) {
+                  // Get the aggregate from references of `If`
+                  val aggRefs = e.aggregateFunction.references.tail.head
```

Comment:
Why cant we simply get them all, this seems convoluted

Reply:
This seems much more fragile than it needs to be.

Reply:
It contains (gid, aggregate). Since I was interested in the  aggregate Attribute, I was doing this way. I could get them all but would discard `gid` later.

Reply:
I understand, thanks for the explanation. We are depending heavily on the order of the attributes in this case which makes an already slightly fragile code to be more susceptible to future breakage. Take all and filter would be more useful here.